models/DataObject/ClassDefinition/Data.php

```diff
@@ -654,6 +654,20 @@ protected function getPreGetValueHookCode(string $key): string
         return $code;
     }
 
+    /**
+     * @return string
+     */
+    public function getInputType() {
```

Comment:
Should we really go for a default here on the abstract? 
Imho it would be better to introduce an `TypeHintSupportInterface` that needs to be implemented by the data-type to support PHP type-hints.